### PR TITLE
Chore: Kill claims store

### PR DESCRIPTION
**Pull request by rasmusguterstam**
(Implementation of [this](https://github.com/HedvigInsurance/hedgehog/pull/270).)

- Kotlinize ClaimResolver
- Implement listClaims

### CODE REVIEW

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.stream.Collectors
+
+@Component
+class ClaimResolver(
+    private val memberLoader: MemberLoader,
+    private val uploadedFilePostprocessor: UploadedFilePostprocessor,
+    private val productPricingService: ProductPricingService
+) : GraphQLResolver<Claim?> {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt:25]:
```suggestion
) : GraphQLResolver<Claim> {
```

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    fun getMember(claim: Claim): CompletableFuture<Member> {
+        return try {
+            memberLoader.load(claim.memberId)
+        } catch (exception: Exception) {
+            CompletableFuture()
+        }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt:31]: Maybe we should logger.error here for potential future debugging

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                UploadedAt,
+                claimId,
+                category,
+                contentType
+            )
+            claimUpload
+        }.collect(Collectors.toList())
+        claimFileUploads.addAll(claimFileUploadDtos)
+        return claimFileUploads
+    }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt:53]: Can make this one a bit neater
```suggestion
    fun getClaimFiles(claim: Claim): List<ClaimFileUpload> {
        if (claimFiles.isEmpty()) {
            return emptyList()
        }
        return claimFiles.stream().map { claimFile ->
            ClaimFileUpload(
                claimFile.claimFileId,
                uploadedFilePostprocessor.processFileUrl(claimFile.key, claimFile.bucket),
                claimFile.UploadedAt,
                claimFile.claimId,
                claimFile.category,
                claimFile.contentType
            )
        }
    }
```

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    fun getType(claim: Claim, env: DataFetchingEnvironment?): Any? {
+        return if (claim._type == null) {
+            null
+        } else when (Util.claimType(claim._type)) {
+            ClaimTypes.TheftClaim -> {
+                TheftClaim.fromClaimData(claim._claimData)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt:60]:
```suggestion
    fun getType(claim: Claim, env: DataFetchingEnvironment?): Any? {
        return when(claim._type)) {
            ClaimTypes.TheftClaim -> {
                TheftClaim.fromClaimData(claim._claimData)
            ....
            else -> null
```

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/GraphQLQuery.java [hunk 5]
```diff
@@ -194,11 +198,24 @@ public MemberSearchResult memberSearch(String query, MemberSearchOptions options
             getToken(env)
         );
 
-        System.out.println(searchResult);
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/resolvers/GraphQLQuery.java:197]: Nice catch

File: src/main/java/com/hedvig/backoffice/graphql/types/ListClaimsResult.kt [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import com.hedvig.backoffice.services.claims.dto.ClaimSearchResultDTO
+
+data class ListClaimsResult(
+    val claims: List<Claim>,
+    val totalPages: Int,
+    val page: Int
+) {
+    companion object {
+        fun from(listClaimsSearchResultDTO: ClaimSearchResultDTO) = ListClaimsResult(
+            claims = listClaimsSearchResultDTO.claims.map { Claim.fromDTO(it) },
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/types/ListClaimsResult.kt:12]:
```suggestion
            claims = listClaimsSearchResultDTO.claims.map((Claim)::fromDTO),
```

**Review by vonElfvin — COMMENTED** [review]
Neat 😍 Added some code improvements now that we use Kotlin :)

File: src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt (hunk 3 above)
Comment (rasmusguterstam) [reply src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt:53]: Nice one!

**Review by vonElfvin — APPROVED** [review]

**Pull request merged**